sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -684,10 +678,15 @@ AllocaCommandBase *Scheduler::GraphBuilder::findAllocaForReq(
 static bool checkHostUnifiedMemory(const ContextImplPtr &Ctx) {
   if (const char *HUMConfig = SYCLConfig<SYCL_HOST_UNIFIED_MEMORY>::get()) {
     if (std::strcmp(HUMConfig, "0") == 0)
-      return Ctx->is_host();
+      return false;
```

Comment:
```suggestion
      return Ctx == nullptr;
```
Pretty sure it should be this instead.